src/target/riscv/riscv-013.c

```diff
@@ -3203,35 +3220,34 @@ static int read_memory_bus_v0(struct target *target, target_addr_t address,
 /**
  * Read the requested memory using the system bus interface.
  */
-static int read_memory_bus_v1(struct target *target, target_addr_t address,
-		uint32_t size, uint32_t count, uint8_t *buffer, uint32_t increment)
+static int read_memory_bus_v1(struct target *target, riscv_mem_access_info_t access)
 {
-	if (increment != size && increment != 0) {
+	if (access.increment != access.element_size && access.increment != 0) {
 		LOG_TARGET_ERROR(target, "sba v1 reads only support increment of size or 0");
 		return ERROR_NOT_IMPLEMENTED;
 	}
 
-	assert(size <= 16);
-	assert(IS_PWR_OF_2(size));
+	assert(access.element_size <= 16);
+	assert(IS_PWR_OF_2(access.element_size));
 
 	dm013_info_t *dm = get_dm(target);
```

Comment:
The code of `read_memory_bus_v1` is already quite complicated (and bugs were discovered here in the past).

For that reason, I'd recommend to make every effort to keep this part readable. Please consider unpacking the structure to individual variables, as it is done in 0.11 versions of read_memory and write_memory. That way the code will become somewhat simpler to read (the reader would not be bothered by all those `access.` prefixes).

Reply:
Addressed